language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE ConstraintKinds      #-}
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE DerivingVia          #-}
+{-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE RankNTypes           #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Language.PlutusCore.Evaluation.Machine.ExMemory
+( Plain
+, WithMemory
+, ExMemory(..)
+, ExCPU(..)
+, GenericExMemoryUsage(..)
+, ExMemoryUsage(..)
+, withMemory
+) where
+
+import           Language.PlutusCore
+import           PlutusPrelude
+
+import           Control.Monad.RWS.Strict
+import qualified Data.ByteString.Lazy     as BSL
+import qualified Data.Text                as T
+import           Foreign.Storable
+
+import           GHC.Generics
+
+{- Note [Memory Usage for Plutus]
+
+The base unit is 'ExMemory', which corresponds to machine words. For primities, we use static values for the size, see the corresponding instances. For composite data types, the Generic instance is used, + 1 for the constructor tag. For ADTs, the currently selected branch is counted, not the maximum value.
```

Comment:
That's a reeeally long line.

Reply:
Yeah, VSCode sucks at justifying :-( I haven't found anything in terms of addons.